### ISSUE TITLE
Won't Launch on macOS 13.5

### ISSUE THREAD

**Issue by hemicyon** [rmcrackan/Libation#712]
**Describe the bug**
The app simply will not launch. When looking at the force quit window and trying to open the app, it appears in the list briefly but instantly disappears.

Opening the app for the first time was expectedly met with Apple's unverified warning, clicked okay and opened the security settings and clicked the open anyway button. Tried to launch the app again, but it fails every time.

**To Reproduce**
Open the app on macOS 13.5, I guess

**Expected behavior**
The app should launch

**Platform**
2019 MacBook Pro (Intel) macOS 13.5

**Log Files**
Not sure where to get the log files if the application won't launch. I found some in the package contents, but not sure if those are the right ones

**Comment by Mbucari** [comment]
Is there a file named `LibationCrash.log` in your user account folder?

If not, is there a `Libation` folder in your user account folder containing a file called something like `Log202308.zip`?

**Comment by hemicyon** [comment]
Nope, no Libation directory or log files like that anywhere unfortunately.

**Comment by Mbucari** [comment]
@hemicyon I'm sorry, but I'm not going to be able to help you. The fact that there are no log files suggests that the app bundle doesn't like your machine. Unfortunately I'm not a mac dev, so I really have no idea where to even begin looking for possible causes.

Really sorry about that :(

**Comment by 1scriptz** [comment]
Yeah, same issue for me as well on MacOS 13.4.

No log files or anything to be seen. Going to keep trying older releases to hopefully find something that'll work and report back.

**Comment by Mbucari** [comment]
@1scriptz So weird. So the app bundle is the problem??

**Comment by 1scriptz** [comment]
> 

It smells of a bundling issue. Either your bundling process is out of date or Apple changed something on their end to cause the problem. They are all the time tweaking things for "Security".

If OP can - try launching the app with my steps I provided to see if it now works on his system as well.

**Comment by Mbucari** [comment]
@1scriptz I just don't understand why I can't reproduce it on my mac mini running macOS 13.4.1. Any ideas?

**Comment by 1scriptz** [comment]
@Mbucari Not a clue 😢

**Comment by hemicyon** [comment]
> Yeah, same issue for me as well on MacOS 13.4.
> 
> No log files or anything to be seen. Going to keep trying older releases to hopefully find something that'll work and report back.
> 
> Edit: Here is what appears to be a json object in the error message that might be part of your error logging system?
> 
> Edit 2: Success! I'm running mac m1 v13.4 with the latest release.
> 
> My steps: I had to right click on the executable file "Libation", Click on "Show Package Contents", open the folder "MacOS", scroll down until you see the executable file again called "Libation", right click again and select open. You might have to do the last part twice but it will successfully open and run the program.
> 
> [error-log.txt](https://github.com/rmcrackan/Libation/files/12387045/error-log.txt)

Yes, following these steps successfully launches the program!

**Comment by hemicyon** [comment]
@Mbucari while I was able to open the program using 1scriptz's steps, I didn't have the time to test if it worked, so I just closed it after confirming it would launch. I noticed today that there is now a crash log, even though I don't think it crashed that day, maybe it could be useful?
[LibationCrash.log](https://github.com/rmcrackan/Libation/files/12447951/LibationCrash.log)

**Comment by Mbucari** [comment]
@hemicyon Thanks for the report, but I don't think that's anything to worry about. It looks like you closed Libation right as it was about to display a message to you, and it was unable to do that after the main Libation window had closed.

---

@1scriptz @hemicyon I've got a new thing for you to try if you're willing.

1. Can you please open the Mac Console app (it's in utilities. Not to be confused with Terminal)
2. Click "Start" to for Console to begin collecting system info.
3. Try to run the Libation app bundle.
4. In Console, search for "Libation" and then share those lines here.

Console looks like this:
![image](https://github.com/rmcrackan/Libation/assets/37587114/b76f9704-3957-4e43-8713-57406fd1f4cf)

**Comment by Mbucari** [comment]
Reopening this issue because there may be hope after all.

**Comment by hemicyon** [comment]
Interestingly enough, the app launched successfully this time. I couldn't see a straightforward way to export the console data, so I tried copying and pasting into a text document, but it doesn't look the same as it's presented in the app.

[libation console lines.txt](https://github.com/rmcrackan/Libation/files/12448411/libation.console.lines.txt)

<img width="1904" alt="Screenshot 2023-08-27 at 12 24 13" src="https://github.com/rmcrackan/Libation/assets/75453443/1bcefa98-ec3c-438a-ba65-b97eafbc323f">

**Comment by Mbucari** [comment]
@hemicyon I learned of a good way to peek at what's going wrong, and the issue resolves itself. Typical quantum bug: observing the behavior changes it.

Still, thanks for trying. And now we have a new troubleshooting step for macOS!

**Comment by 1scriptz** [comment]
Yeah, quantum bug - opens no issue now.

I also went and deleted the Libation folder out of my account folder. It still fired right up without a issue. Spooky. 

Possibly there is another file created apart from the libation folder at initial start?

**Comment by Mbucari** [comment]
> Possibly there is another file created apart from the libation folder at initial start?

Yes. On first launch libation creates a file called appsettings.json in the program files directory. [There was a recent issue where this process was described in detail.](https://github.com/rmcrackan/Libation/issues/725)

I've tested that and it does work on my machine because, but we've already established something is different between my machine and your machine so this might be the culprit.

Can you please try something for me?

1. Move the Libation.app bundle from your applications directory to some other temp location.
2.  Download a fresh copy of the v11.0.1 and extract it to your applications folder.
3. Run the gatekeeper exception command from the [MacOS installation instructions page](https://github.com/rmcrackan/Libation/blob/master/Documentation/InstallOnMac.md). At this point, if the problem is indeed the missing appsettings.json then Libation should fail to open. If it does open at this step, please report back. If it doesn't, continue to step 4.
4. Go into the app bundle that you moved to a temporary location in step 1. Copy ./Contents/MacOS/appsettings.json to the same place inside the fresh Libation.App in your applications folder.
5. Re-run the gatekeeper command from step 3. If that was the problem, libation should now run.

**Comment by Mbucari** [comment]
@1scriptz @hemicyon 
> Can you please try something for me?

Pwease?! 🥹

I'd really like an answer to this so we can improve the Libation experience for everyone else moving forward.

**Comment by Mbucari** [comment]
> I believe it should first check and create Libation folder? Otherwise the writealltext just failed un-intendly.

I noticed that and have already [added a fix](https://github.com/rmcrackan/Libation/commit/f86c77a5463bc5585a42260e035710bfa78e415d) that will be included in the next release.

But this issue may or may not be related to appsettings.json. Libations seems to work out of the box on macOS for most people, but there are a few who can't open it. The best idea right now its that their problem relates to appsettings.json, but I can't confirm that unless _somebody_ runs the experiment.

**Comment by hemicyon** [comment]
@Mbucari in my case, Libation launched at step 3

**Comment by Mbucari** [comment]
> @Mbucari in my case, Libation launched at step 3

I'm not terribly surprised, but I am disappointed. Would you mind trying one more thing? Do steps 1 and 2, but also delete or rename the `Libation` folder in your user profile folder, then try running libation.

**Comment by hemicyon** [comment]
> > @Mbucari in my case, Libation launched at step 3
> 
> I'm not terribly surprised, but I am disappointed. Would you mind trying one more thing? Do steps 1 and 2, but also delete or rename the `Libation` folder in your user profile folder, then try running libation.

Done and done. Although, perplexingly, Libation again launched without fuss at step 3

**Comment by Mbucari** [comment]
> Done and done. Although, perplexingly, Libation again launched without fuss at step 3

Thanks for all your work, @hemicyon. It's really unfortunate we weren't able to identify the cause, but at lease we now have a troubleshooting step for mac users.
